app/controllers/businessactivities/RiskAssessmentController.scala

```diff
@@ -59,21 +59,21 @@ class RiskAssessmentController @Inject() (val dataCacheConnector: DataCacheConne
 
             for {
               businessActivities <- dataCacheConnector.fetch[BusinessActivities](BusinessActivities.key)
-              _ <- dataCacheConnector.save[BusinessActivities](BusinessActivities.key, businessActivities.riskAssessmentPolicy(data))
-            } yield redirectDependingOnEdit(edit, ControllerHelper.isAccountancyServicesSelected(Some(businessMatching)))
-
+              _ <- dataCacheConnector.save[BusinessActivities](BusinessActivities.key, businessActivities.riskAssessmentHasPolicy(data))
+            } yield redirectDependingOnEdit(edit, ControllerHelper.isAccountancyServicesSelected(Some(businessMatching)), data)
           }
         } recoverWith {
           case _: IndexOutOfBoundsException => Future.successful(NotFound(notFoundView))
         }
       }
   }
 
-  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean) = edit match {
-    case true => Redirect(routes.SummaryController.get())
-    case false => accountancyServices match {
-      case true => Redirect(routes.SummaryController.get())
-      case false => Redirect(routes.AccountantForAMLSRegulationsController.get())
+  private def redirectDependingOnEdit(edit: Boolean, accountancyServices: Boolean, data: RiskAssessmentHasPolicy) =
```

Comment:
here might be necessary some refactor, first and third case for example.

Reply:
Done

Reply:
Removed 3rd statement.